#T_base_sensor:
# - [1, 0, 0, 0.3]
# - [ 0, 1, 0, 0]
# - [ 0,  0, 1, 0]
# - [ 0.0, 0.0, 0.0, 1.0]
 
# General Params
verbosity: 3
label_path: /home/lukas/Documents/PanopticMapping/Data/flat_dataset/labels.csv
global_frame_name: world
change_detection_interval: 2.0

tsdf_integrator:  
  type: projective # naive, projective, projective_multi_tsdf
  verbosity: 1
  
  # projective args
  width: 640
  height: 480
  vx: 320
  vy: 240
  focal_length: 320
  
  max_range: 5  # m
  min_range: 0.1
  use_weight_dropoff: true
  sparsity_compensation_factor: 1.0
  max_weight: 10000
  interpolation_method: semantic  # nearest, bilinear, adaptive, semantic
  foreign_rays_clear: true
  
  # naive args
  voxblox_integrator_type: fast # projective, fast
  sensor_vertical_field_of_view_degrees: 73.7397952
  use_const_weight: true  
  max_ray_length_m: 5
  voxel_carving_enabled: true
  use_weight_dropoff: true
  sparsity_compensation_factor: 1.0

#  truncation_distance: 0.2
#  max_ray_length_m: 5.0
#  use_const_weight: false
#  use_sparsity_compensation_factor: true
#  sparsity_compensation_factor: 20.0
#  integration_order_mode: "sorted"

id_tracker:
  type: ground_truth
  verbosity: 2
  instance_voxel_size: 0.03
  background_voxel_size: 0.08
  unknown_voxel_size: 0.08
  freespace_voxel_size: 0.3
  voxels_per_side: 16
  
visualization:
  visualization_interval: 1.0
  visualization_mode: submaps # submaps, instances, color, normals, classes, change
  submap_color_discretization: 20
  visualize_mesh: true
  visualize_tsdf_blocks: false
  visualize_free_space: true
  include_free_space_in_meshes: false
  

